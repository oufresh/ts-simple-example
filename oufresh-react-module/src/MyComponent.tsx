import * as React from "react";

export interface IMyComponentProps {
    text: string;
};

export const MyComponent = (props: IMyComponentProps) => {
    const str = "Hello, the text is: " + props.text;
    return <div><label>{str}</label></div>;
};
