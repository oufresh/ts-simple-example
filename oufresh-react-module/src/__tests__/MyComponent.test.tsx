import * as React from "react";
import {
  render,
  cleanup
  /*fireEvent,
  cleanup,
  waitForElement*/
} from "@testing-library/react";
import "jest-dom/extend-expect";
import { MyComponent } from "../MyComponent";

afterEach(cleanup);

test("Test MyComponent", () => {
  const { asFragment, getByText } = render(<MyComponent text={"PROVA"} />);
  expect(asFragment()).toMatchSnapshot();
  expect(getByText("Hello, the text is: PROVA")).not.toBeUndefined();
});
