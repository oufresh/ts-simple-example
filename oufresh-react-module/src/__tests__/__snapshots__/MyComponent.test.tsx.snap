// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyComponent 1`] = `
<DocumentFragment>
  <div>
    <label>
      Hello, the text is: PROVA
    </label>
  </div>
</DocumentFragment>
`;
